qiskit/circuit/library/standard_gates/multi_control_rotation_gates.py

```diff
@@ -81,6 +83,106 @@ def _apply_mcu_graycode(circuit, theta, phi, lam, ctls, tgt, use_basis_gates):
         last_pattern = pattern
 
 
+def mcsu2_real_diagonal(
```

Comment:
Maybe it would be better to mark this function as private as it is a utility and not API documented. If we want to expose it, it might be worth to turn it into it's own class in a follow-up PR.

Reply:
I kept it public, but we can easily make its private and delete the release note of #9688.

Reply:
I think it would be better if it's private, this way we can still move it e.g. to the synthesis module later, or turn it into a class without having to deprecate it. Since the release is imminent, I'll open a short separate PR for that 🙂